core/PegasusCore.cpp

```diff
@@ -296,10 +303,16 @@ namespace pegasus
         }
     }
 
+    // This method will execute a single thread for up to X instructions
+    // where X is the quantum for that thread. The cycle count of all threads
+    // is updated to match the current thread's cycle count. Then this event
+    // is rescheduled at that cycle count.
     void PegasusCore::advanceSim_()
     {
         PegasusState* state = threads_[current_hart_id_];
-        if (state->getSimState()->sim_stopped == false)
+        bool pause_thread = false;
+        if ((state->getSimState()->sim_stopped == false)
```

Comment:
Might be worth making a local variable to cache the simstate since `state->getSimState()` is called a lot.